### PR TITLE
modernize mkppport and lib/unicore/mktables

### PR DESCRIPTION
**Pull request by atoomic**
use function signatures in lib/unicore/mktables and mkppport avoiding some boilerplates to check for @_.

### CODE REVIEW

File: lib/unicore/mktables [hunk 1]
```diff
@@ -8168,14 +7909,10 @@ sub trace { return main::trace(@_); }
         return;
     }
 
-    sub is_set_equivalent_to {
+    sub is_set_equivalent_to($self, $other=0) {
```
Comment (khwilliamson) [inline lib/unicore/mktables]: I believe $other=undef

File: lib/unicore/mktables [hunk 2]
```diff
@@ -8620,10 +8339,6 @@ sub trace { return main::trace(@_); }
 
         return unless $debugging_build;
 
-        my $leader = shift;   # Should only be called on the leader table of
```
Comment (khwilliamson) [inline lib/unicore/mktables:8623]: The comment shouldn't go away

File: lib/unicore/mktables [hunk 3]
```diff
@@ -9657,12 +9334,7 @@ sub trace { return main::trace(@_) if main::DEBUG && $to_trace }
         return;
     }
 
-    sub set_fate {
-        my $self = shift;
-        my $fate = shift;
-        my $reason = shift;  # Ignored unless suppressing
```
Comment (khwilliamson) [inline lib/unicore/mktables:9663]: $reason is not optional, and the comment should be retained

File: lib/unicore/mktables [hunk 4]
```diff
@@ -9762,7 +9434,7 @@ sub display_chr {
     return "\x{25CC}$chr";
 }
 
-sub join_lines($) {
+sub join_lines {
```
Comment (khwilliamson) [inline lib/unicore/mktables]: There is a shift in this sub that you could name

File: lib/unicore/mktables [hunk 5]
```diff
@@ -10088,7 +9743,7 @@ sub Standardize($) {
     return $name;
 }
 
-sub standardize ($) {
+sub standardize ($str) {
```
Comment (khwilliamson) [inline lib/unicore/mktables]: You left the shift in in this sub

**Review by khwilliamson — CHANGES_REQUESTED** [review]
Why do you have to reproclaim strict, etc in every package?

File: lib/unicore/mktables (hunk 1 above)
Comment (atoomic) [reply lib/unicore/mktables]: right fixing it

File: lib/unicore/mktables (hunk 2 above)
Comment (atoomic) [reply lib/unicore/mktables:8623]: restoring it

File: lib/unicore/mktables (hunk 3 above)
Comment (atoomic) [reply lib/unicore/mktables:9663]: comment restored
nothing was enforcing `$reason` to be a mandatory argument, and it was possible before this change to call it without a reason, I can introduce a behavior change and make it mandatory

File: lib/unicore/mktables (hunk 4 above)
Comment (atoomic) [reply lib/unicore/mktables]: indeed good idea

File: lib/unicore/mktables (hunk 5 above)
Comment (atoomic) [reply lib/unicore/mktables]: indeed and need to set the default to undef, as the function return undef on undef input.. 
fixing it

File: lib/unicore/mktables (hunk 3 above)
Comment (atoomic) [reply lib/unicore/mktables:9663]: `Too few arguments for subroutine 'Property::set_fate' at lib/unicore/mktables line 9176.` in `sub set_proxy_for` is calling it with a single arg

Comment (atoomic) [reply lib/unicore/mktables:9663]: restoring it as an optional argument for now

**Comment by atoomic** [comment]
@khwilliamson fbdf3dd5 is providing the changes requested during the first review, let smoke it before merging changes can probably be spotted from 1e192bb96307affeb51adbbce821ae7d32469cf5 [ which I had to modify later to fix mktable ]

File: lib/unicore/mktables (hunk 3 above)
Comment (khwilliamson) [reply lib/unicore/mktables:9663]: I thought I'd checked during my review that it was always present, but if not, ok

**Pull request merged**